### ISSUE TITLE
Work out access rights for post and comment creation

### ISSUE THREAD

**Issue by begedin** [code-corps/deprecated-code-corps-rails-api#50]
- When should the user be allowed to create posts for a project?
  - Is it `post_type` specific? Role-specific? Project type specific? We need to flesh this out.
- When should the user be allowed to create comments for a post? 
  - Some overlap with posts here, but also differences, I think.

**Comment by joshsmith** [comment]
I think this is a combination of role inside a project and the post type.

I think comments on posts will always be allowed, regardless of the post type or relationship to a project.

Issues and ideas can be posted by anyone.

Progress posts and tasks can only be created by contributors to a project. This should encourage a higher level of ownership.

Also, the poster can close any post they created, but posts can be closed by project admins at will, though not contributors.

In this sense, the hierarchy is:
- Owner
- Admins
- Contributors
- Everyone

**Comment by begedin** [comment]
This makes sense, but we there's no point in partially implementing it, until we have contributors. Blocked by #12

**Comment by joshsmith** [comment]
This was done for posts in #103 but we need to do it for comments. Creating a separate issue though.
